backend/app/utils/decorators.py

```diff
@@ -0,0 +1,27 @@
+"""Custom decorators for role-based access control."""
+from functools import wraps
+from flask_jwt_extended import verify_jwt_in_request, get_jwt_identity
+from flask import abort
+from ..models.user import User
+from ..extensions import DB as db
+
+def roles_required(*required_roles):
+    """Decorator to check if a user has one of the required roles."""
+
+    def decorator(f):
```

Comment:
I recommend you to follow the following steps for this specific middleware like code.  

- [ ] Define a new colder called middlwares
- [ ] Define a new class called is_admin.py and define is_admin decorator and implement a logic to check if the give user is an admin or not. Don't pass any param to your decorator, technically accepting param means opening a door for manipulation.
- [ ] do the same for other checks like is_editor.py and is_regular.py. A file per middlware logic to keep it simple and separate concerns. We might need other middlwares in the future and it's better to keep this in mind.